tools/envoy_headersplit/headersplit.py

```diff
@@ -0,0 +1,367 @@
+# !/usr/bin/env python3
+# Lint as: python3
+"""
+This python script can dividing monolithic mock headers
+into different mock classes. We need to remove the
+over-included head files in generated class codes and
+resolve dependencies in the corresponding Bazel files
+manually.
+"""
+
+from __future__ import print_function
+
+import argparse
+from typing import Type, List, Tuple, Dict
+import clang.cindex
+from clang.cindex import TranslationUnit, Index, CursorKind, Cursor
+
+clang.cindex.Config.set_library_path("/opt/llvm/lib")
+
+
+def to_filename(classname: str) -> str:
+  """
+    maps mock class name (in C++ codes) to filenames under the Envoy naming convention.
+    e.g. map "MockAdminStream" to "admin_stream"
+
+    Args:
+        classname: mock class name from source
+
+    Returns:
+        corresponding file name
+    """
+  filename = classname.replace('Mock', '', 1)  # Remove only first "Mock"
+  ret = ""
+  for index, val in enumerate(filename):
+    if val.isupper() and index > 0:
+      ret += '_'
+    ret += val
+  return ret.lower()
+
+
+def get_headers(translation_unit: Type[TranslationUnit]) -> str:
+  """
+    extracts all head includes statements from the target code file (translation_unit)
+
+    for instance:
+        foo.h:
+        #include "a.h"
+        #include "b.h"
+
+        int foo(){
+
+        }
+    this function should return
+    '#include "a.h"\n#include "b.h"'
```

Comment:
This is an unusual API. It makes sense in this case, since we're pasting the output into a new file, but it's still surprising. Can we document why we're returning a string instead of a list here?

Reply:
Make the libclang get_includes API work will need extra effort than necessary since we only need to paste the include directives in new files. The reason of returning string instead of list, is that would be easier to implement as we are going to paste the result to new file. Also the string will contains other directives like #pragma in the origin code.
I have added the comments explaining why returning string in the script